### ISSUE TITLE
Alias `afterRender` to `afterEachRender`

### ISSUE THREAD

**Issue by mark-akturatech** [angular/angular#54825]
### Which @angular/* package(s) are relevant/related to the feature request?

@angular/core

### Description

@angular/core `afterRender` and `afterNextRender` are great however it is too easy to miss the distinction during a peer review and this can have a big effect on performance.

Is it possible to alias `afterRender` to `afterEachRender` or similar (I understand that it is too late to change names now). 
The new name would be clear that the code attached will run multiple times (as opposed to once only if using afterNextRender).

### Proposed solution

Rename `afterRender` to `afterEachRender`

### Alternatives considered

Currently if used I asked the devs to `import { afterRender as afterEachRender }`. We could add a lint task to ensure this is done.

BTW I don't get a chance to say this elsewhere - so will say it here - thank you for all your work. Every change your team makes is so positive.

**Comment by eneajaho** [comment]
I'd go with afterEachRefresh -> as Angular refreshes bindings and doesn't re-render.

**Comment by robertIsaac** [comment]
> I always wondered why `AfterNextRender` is not named something like `AfterFirstRender` or something.
> 
> It's very confusing to distinguish between the AfterNextRender and AfterRender, my colleague keeps forgetting what is what - with quite big impact.

true, same here

**Comment by hiepxanh** [comment]
Jest it very confuse, I think there is an action we doing after something, like an action layer on other layer, so we can use `on`, like:
- `onRender(( ) => { })` : everytime we render it, no matter where it is server or client
- `onFirstRender(( ) => { }) `: the first time we render

**Comment by mauriziocescon** [comment]
Same problem here... 

Progressively replacing `ngAfterViewInit` with `afterNextRender` (following https://github.com/angular/angular/discussions/49682 without even any SSR). 

It's honestly a bit of a mess with the current names... combining: speed, IDE autocomplete capabilities, right function call is almost an NP-Hard problem! 😅

IMHO, too much error prone!

**Comment by JeanMeche** [comment]
The team agrees that the name could be improved. The change will likely land as part of the stabilisation process of these 2 functions.

**Comment by JeanMeche** [comment]
We decided to removed `afterRender`, see #60794.

**Comment by mauriziocescon** [comment]
@JeanMeche say: still `afterNextRender` is far from an ideal name. That `next` makes no sense IMHO... except for confusing devs. 😅

Just an idea: how about a migration doing the following
- rename `afterRender` to `__afterRender`, 
- then rename `afterNextRender` to `afterRender` / `afterFirstRender` / ... ?

In the end, they are both in dev-preview!

**Comment by pkozlowski-opensource** [comment]
@mauriziocescon we can debate names but we shouldn't use anything that suggests "first" / "initial" - the `afterNextRender` hook executes in the _next_ render cycle _after_ it was created. Naming things is hard, but the current name exactly reflect what this hook does...

**Comment by mauriziocescon** [comment]
Well, I understand... just saying it can be easily misunderstood or not understood at all (especially if you come from the `onMount(ed)` world 😅). 

But ok... it can be a point for the future as well!

**Comment by pkozlowski-opensource** [comment]
@mauriziocescon I think there is a misunderstanding here - `afterNextRender` is _not_ an equivalent of mount-like hooks (again insisting on the fact that this is _not_ about the initial render!). `afterNextRender` can happen _any_ time in the component's lifetime - what matters is the timing of the hook's creation and change detection.

**Comment by JeanMeche** [comment]
+1, was about to answer you on twitter. 


`afterNextRender` is a global hook. 
It fires after the CD process. 

This is done for a reason, it's mainly a DOM hook you want to call once the DOM has stabilized

**Comment by mauriziocescon** [comment]
Ok, I might be completely off then... sorry for the confusion! 

Let me explain how I (rarely) use it then: as an OnInit that runs only on the client without phases! 
```ts
// onInit called client-side only and potentially doing DOM manipulation 
private readonly domReady = afterNextRender(() => this.someInit());
```
I must admit I've never took a deep look at the phases part. My bad! 

I wonder: would you be so kind to elaborate on what I'm doing wrong? Or correct my mental model regarding it? 🙏

**Comment by JeanMeche** [comment]
If it's for DOM manipulation it's fine. 

The phases are here to prevent multiple reflow by ordering the multiple after render callbacks. earlyRead > Write > ReadWrite > Read.

**Comment by mauriziocescon** [comment]
Yeah... that part was more or less clear (again haven't checked all the details... and I'm not a real expert of reflows)! 

What I wonder is: maybe designed with a different purpose in mind... but isn't `afterNextRender(() => this.someInit())` logically equivalent to an `onMonted`?

**Comment by JeanMeche** [comment]
`onMounted` is more like the first invocation of `AfterViewChecked` which is scoped to the rendering of the component.

**Comment by mauriziocescon** [comment]
Say: what `onMounted` ([solid](https://docs.solidjs.com/reference/lifecycle/on-mount) / [vue](https://vuejs.org/api/composition-api-lifecycle#onmounted) / [svelte](https://svelte.dev/docs/svelte/lifecycle-hooks#onMount)) does is clear enough to me! And that's mostly what I'd like to use. 

What is not clear to me is the relationship between `afterNextRender` and `onMounted`. 
I see 2 outcomes: 
1. `afterNextRender` offers more possibilities than `onMounted` (and you should learn about those), but it can be used as `onMounted` (more or less logically equivalent), 
2. `afterNextRender` is totally different compared to `onMounted` and you're abusing it; see this counterexample for `afterNextRender(() => this.someInit())` where __ this_bug __ happens!

**Comment by JeanMeche** [comment]
`afterNextRender` somehow offers more possibilities. 

Imagine you want to read the size of your component, but that size directly depends on the size of a neighbouring component (like with flex ratios). 
Reading the size in `afterViewCheck` is too early because it'll change after when the neighbouring component is rendered. Reading it after everything is rendered, is actually the best time for it. 
And since you might have several DOM apis that read/write the DOM it's important to have them ordered to prevent unnecessary reflows.

**Comment by mauriziocescon** [comment]
Ok, I checked the [phases part in the doc](https://angular.dev/api/core/afterNextRender) and indeed it's clear the reason of the name now! Thanks! 

Say: maybe it's just me, but I have the feeling it targets sort of lib / advance cases... rather than more common ones in pure DEV land "hey I need to do some basic DOM manipulation (elRef) and I want it to be safe while using SSR (or even just future proof)". 

The fact all phases are optional doesn't help either (yeah... more APIs then). 
So I got confused! 😵‍💫

Anyway: thanks!
